examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,399 @@
+                llm=llm_config,
+                name=name,
+                system_message=system_message,
+            )
+        )
+
+    pro_agent = create_chat_agent("Pro",
+                                  pro_agent_config,
+                                  system_messages.messages[pro_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    con_agent = create_chat_agent("Con",
+                                  con_agent_config,
+                                  system_messages.messages[con_key].message +
+                                  DEFAULT_SYSTEM_MESSAGE_ADDITION)
+    feedback_agent = create_chat_agent("Feedback",
+                                       feedback_agent_config,
+                                       FEEDBACK_AGENT_SYSTEM_MESSAGE)
+    logger.info("Pro, Con, and feedback agents created.")
+
+    # Determine user's side and assign user_agent and ai_agent based on user selection
+    agents = {"pro": (pro_agent, con_agent, "Pro", "Con"), "con": (con_agent, pro_agent, "Con", "Pro")}
+    user_agent, ai_agent, user_side, ai_side = agents[side]
+    logger.info(
+        f"Starting debate on topic: {topic_name}, taking the {user_side} side. "
+        f"LLM Delegate: {llm_delegate}"
+    )
+
+    # Clear Agents memory
+    user_agent.clear_history()
+    ai_agent.clear_history()
+    logger.info(f"\n{user_side} Agent ({topic_name}):\n")
+
+    # Determine if the debate is autonomous or the user input for one side
+    if llm_delegate:
+        logger.info("Autonomous Debate Selected")
+        interactive_setting = False
+    else:
+        logger.info("Manual Debate Selected with an AI Agent")
+        interactive_setting = True
+        user_input: str = Prompt.ask(
+            "Your argument (or type 'f' for feedback, 'done' to end):"
+        )
+        user_agent.llm = None  # User message without LLM completion
+        user_agent.user_message = user_input
+
+    # Set up langroid tasks and run the debate
+    user_task = Task(user_agent, interactive=interactive_setting, restart=False)
+    ai_task = Task(ai_agent, interactive=False, single_round=True)
+    user_task.add_sub_task(ai_task)
+    user_task.run("get started", turns=max_turns)
+
+    # Determine the last agent based on turn count and alternation
+    # Note: user_agent and ai_agent are dynamically set based on the chosen user_side
+    last_agent = ai_agent if max_turns % 2 == 0 else user_agent
+
+    # Generate feedback summary and declare a winner using feedback agent
+    validation_message: List = last_agent.message_history
+    if not validation_message:
+        logger.warning("No agent message history found for the last agent")
+        validation_message = "No last agent history available to validate."
+
+    feedback_agent.clear_history()
+    feedback_agent.enable_message(MetaphorSearchTool)
+    feedback_agent_task = Task(feedback_agent, interactive=False)
+    feedback_agent_task.run(validation_message)
```

Comment:
Since there is no clear done-condition, this gets stuck in a loop, and exits after 5 steps. The `validation_message` must include instructions to say DONE after Step 3.

Reply:
done